app/src/components/objectStorageService.js

```diff
@@ -0,0 +1,283 @@
+        });
+      });
+    } catch (e) {
+      errorToProblem(SERVICE, e);
+    }
+  }
+
+  async uploadFile(prefix, filePath, fileName) {
+    try {
+
+      const fileId = uuidv4();
+      const fileContent = fs.readFileSync(filePath);
+      const key = this._join(this._key, prefix, fileId);
+
+      const params = {
+        Bucket: this._bucket,
+        Key: key,
+        Body: fileContent,
+        ContentType: mime.contentType(path.extname(fileName)),
+        Tagging: `filename=${fileName},id=${fileId}`,
+        Metadata: {
+          'name': fileName,
+          'id': fileId
+        }
+      };
+
+      return new Promise((resolve, reject) => {
+        // eslint-disable-next-line no-unused-vars
+        this._s3.upload(params, (err, data) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve({
+              fileId: fileId,
+              Key: data.Key
+            });
+          }
+        });
+      });
+    } catch (e) {
+      errorToProblem(SERVICE, e);
+    }
+  }
+
+  async copyFile(fileId, from, to) {
+    try {
+      let sourcePath = this._join(from);
+      let destPath = this._join(to);
+
+      const params = {
+        Bucket: `${this._bucket}/${this._key}${destPath}`,
+        CopySource: `${this._bucket}/${this._key}${sourcePath}${fileId}`,
+        Key: fileId
+      };
+
+      return new Promise((resolve, reject) => {
+        // eslint-disable-next-line no-unused-vars
+        this._s3.copyObject(params, (err, data) => {
+          if (err) {
+            reject(err);
+          } else {
+            resolve({
+              fileId: fileId
+            });
+          }
+        });
+      });
+    } catch (e) {
+      errorToProblem(SERVICE, e);
+    }
+  }
+
+  async moveFile(fileId, from, to) {
+    try {
+      let sourcePath = this._delimit(this._join(from));
+      const file = await this.copyFile(fileId, from, to);
+      if (file) {
+        const params = {
+          Bucket: this._bucket,
+          Key: `${this._key}${sourcePath}${fileId}`
+        };
+
+        return new Promise((resolve, reject) => {
+          // eslint-disable-next-line no-unused-vars
+          this._s3.deleteObject(params, (err, data) => {
+            if (err) {
+              reject(err);
+            } else {
+              resolve({
+                fileId: fileId
+              });
+            }
+          });
+        });
+      }
+    } catch (e) {
+      errorToProblem(SERVICE, e);
+    }
+  }
+}
+
+const endpoint = config.get('objectStorage.endpoint');
+const bucket = config.get('objectStorage.bucket');
+const key = config.get('objectStorage.key');
+const accessKeyId = config.get('objectStorage.accessKeyId');
+const secretAccessKey = config.get('objectStorage.secretAccessKey');
+
+let objectStorageService = new ObjectStorageService({accessKeyId: accessKeyId, secretAccessKey: secretAccessKey, endpoint: endpoint, bucket: bucket, key: key});
+module.exports = objectStorageService;
```

Comment:
This could probably just be collapsed into a one liner to avoid having to do another object assignment.